sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+Because this functionality is a common requirement, Azure Core provides a retry policy that can watch for transient failures and automatically retry the request.
+
+This retry policy, therefore, splits the whole pipeline into two parts:
+policies that execute before the retry policy and policies that execute after the retry policy.
+Policies added before the retry policy execute only once per API operation, and policies added after the retry policy execute as many times as the retries.
+
+So, when building the HTTP pipeline, you should understand whether to execute a policy for each request retry or once per API operation.
+
+![](https://learn.microsoft.com/en-us/azure/developer/java/sdk/media/http-pipeline.svg)
+
+#### Common HTTP pipeline policies
+
+HTTP pipelines for REST-based services have configurations with policies for authentication, retries, logging, telemetry, and specifying the request ID in the header.
+These policies are automatically added into HTTP pipeline.
+
+#### Custom HTTP pipeline policy
+
+The HTTP pipeline policy provides a convenient mechanism to modify or decorate the request and response.
+You can add custom policies to the pipeline that the user or the client library developer created.
+When adding the policy to the pipeline, you can specify whether this policy should be executed per-call or per-retry.
+
+To create a custom HTTP pipeline policy, you just inherit from a base policy class and implement some virtual functions.
+You can then plug the policy into the pipeline.
+
+#### Custom headers in HTTP requests
+
+Below is an example of adding a custom header into each HTTP request.
+The header value is static and doesn't change over time, so we make it a per-operation policy.
+If you want to add some time-variant headers like authentication, you should use per-retry policy.
+
+```C++
+class NewPolicy final : public Azure::Core::Http::Policies::HttpPolicy {
+public:
+  ~NewPolicy() override {}
+
+  std::unique_ptr<HttpPolicy> Clone() const override
+  {
+    return std::make_unique<NewPolicy>(*this);
+  }
+
+  std::unique_ptr<Azure::Core::Http::RawResponse> Send(
+      Azure::Core::Http::Request& request,
+      Azure::Core::Http::Policies::NextHttpPolicy nextPolicy,
+      Azure::Core::Context const& context) const override
+  {
+    request.SetHeader("x-ms-custom-hearder-name", "header-value");
+    return nextPolicy.Send(request, context);
+  }
+};
+
+BlobClientOptions options;
+options.PerOperationPolicies.push_back(std::make_unique<NewPolicy>());
+options.PerRetryPolicies.push_back(std::make_unique<NewPolicy>());
+```
+
+## Will the SDK retry failed requests? What is the default retry logic? How can I customize retry behavior?
+
+Requests failed due to network errors or HTTP status code 408, 500, 502, 503, 504 will be retried at most 3 times (4 attempts in total) using exponential backup with jitter.
```

Comment:
Will the list of codes you mentioned here ever change in the future and be risk of getting outdated/incomplete?
```suggestion
Requests failed due to network errors or HTTP status code 408, 500, 502, 503, 504 will be retried at most 3 times (4 attempts in total) using exponential backoff with jitter.
```